src/tensor.c

```diff
@@ -133,20 +179,28 @@ size_t RAI_TensorGetDataSize(const char* dataTypeStr){
 
 void RAI_TensorFree(RAI_Tensor* t){
   if (--t->refCount <= 0){
-    RedisModule_Free(t->tensor.shape);
-    RedisModule_Free(t->tensor.data);
+    if (t->tensor.manager_ctx && t->tensor.deleter) {
+      t->tensor.deleter(&t->tensor);
+    }
+    else {
+      RedisModule_Free(t->tensor.dl_tensor.shape);
```

Comment:
I suggest to switch to a macro of alloc and free and when compile for production point macro to use RedisModule_Free and when compile for testing point macro to use std malloc and free, this way it will be easy to run valgrind on it and test for leaks.

Reply:
Look how we do it in Redisearch

Reply:
Leaks is something we should definitely test at this point, I'll do that before merging

Reply:
This is it right? https://github.com/RedisLabsModules/RediSearch/blob/master/src/rmutil/alloc.h

Reply:
yes and you should call RMUTil_InitAlloc if REDIS_MODULE_TARGET is not defined so it will switch the pointers

Reply:
Where do you suggest is the best place to call it?

Reply:
at the init function?

Reply:
Done